json/src/main/scala/io/sphere/json/package.scala

```diff
@@ -77,6 +77,7 @@ package object json extends Logging {
     default: Option[A] = None
   )(jObject: JObject)(implicit jsonr: FromJSON[A]): JValidation[A] = {
     val fields = jObject.obj
+    // Perf note: avoiding Some(f) with fields.indexWhere and then constant time access is not faster
```

Comment:
I thought removing the `Some(f)` would have an impact but it is completely invisible.

I tried the following

```
 def field[A](
    name: String,
    default: Option[A] = None
  )(jObject: JObject)(implicit jsonr: FromJSON[A]): JValidation[A] = {
    val fields = jObject.obj
    val index = fields.indexWhere(f => f._1 == name && f._2 != JNull && f._2 != JNothing)
    if (index == -1) {
      default.map(Valid(_))
        .orElse(jsonr.read(JNothing).fold(_ => None, x => Some(Valid(x)))) // orElse(jsonr.default)
        .getOrElse(Invalid(NonEmptyList.one(JSONFieldError(name :: Nil, "Missing required value"))))
    } else {
      jsonr.read(fields(index)._2).leftMap(
        errs => errs map {
          case JSONParseError(msg) => JSONFieldError(name :: Nil, msg)
          case JSONFieldError(path, msg) => JSONFieldError(name :: path, msg)
        })
    }
  }
}
```

I also tried to change from repetitive `List[Field]` traversal to a `Map[String, Field]` and it is much slower for small JSON object due to the cost of the `Map`.